File: backend-nestjs-reference/src/health/health.controller.ts
import { Controller, Get } from '@nestjs/common';
import { Public } from '../common/decorators/public.decorator';

@Controller('health')
export class HealthController {
  @Get()
  @Public()
  checkHealth() {
    return { status: 'ok', ts: new Date().toISOString() };
  }
}

File: backend-dotnet/src/API/Controllers/HealthController.cs
using Microsoft.AspNetCore.Mvc;

namespace ScholarPrep.API.Controllers;

[ApiController]
[Route("api/[controller]")]
public class HealthController : ControllerBase
{
    // This method will handle: GET /api/health
    [HttpGet]
    public IActionResult CheckHealth()
    {
        // It just returns a 200 OK with a simple message
        return Ok(new { Status = "Healthy" });
    }
}